### DISCUSSION TITLE
SocketException being thrown

### DISCUSSION THREAD

**Discussion by kallencst** [rabbitmq/rabbitmq-stream-dotnet-client#212] [Q&A] [answered]
I am trying to get a simple example of the dotnet client code to work. I have done the following:
* Downloaded the rabbitmq-stream-dotnet-client source code
* Opened the .sln file in Visual Studio 2022
* Added a new Console App project
* Pasted in the sample code from the "Getting Started" section and added necessary usings.
Now when trying to run this, the console application hangs at the line of code that calls Producer.Create. I have found that it throws SocketException in Connection.Create; the exception message is "An attempt was made to access a socket in a way forbidden by its access permissions." Apparently this is re-tried in an infinite loop, resulting in the hang-up.
The endpoint in Connection looks like this: {[fe80::164b:39b9:2abf:f2a7%7]:5552}
I have tried running Visual Studio as Administrator, and disabling Windows Firewall; this had no effect.

The full stack trace of the exception:
"   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)\r\n   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16 token)\r\n   at System.Threading.Tasks.ValueTask.ValueTaskSourceAsTask.<>c.<.cctor>b__4_0(Object state)\r\n--- End of stack trace from previous location ---\r\n   at RabbitMQ.Stream.Client.Connection.<Create>d__17.MoveNext() in C:\\Users\\kent.allen\\OneDrive - CentralSquare Technologies\\Desktop\\RabbitMQ\\rabbitmq-stream-dotnet-client-main\\RabbitMQ.Stream.Client\\Connection.cs:line 66\r\n   at RabbitMQ.Stream.Client.Client.<Create>d__45.MoveNext() in C:\\Users\\kent.allen\\OneDrive - CentralSquare Technologies\\Desktop\\RabbitMQ\\rabbitmq-stream-dotnet-client-main\\RabbitMQ.Stream.Client\\Client.cs:line 201\r\n   at RabbitMQ.Stream.Client.Routing.<CreateClient>d__4.MoveNext() in C:\\Users\\kent.allen\\OneDrive - CentralSquare Technologies\\Desktop\\RabbitMQ\\rabbitmq-stream-dotnet-client-main\\RabbitMQ.Stream.Client\\RoutingClient.cs:line 26\r\n   at RabbitMQ.Stream.Client.RoutingHelper`1.<LookupConnection>d__0.MoveNext() in C:\\Users\\kent.allen\\OneDrive - CentralSquare Technologies\\Desktop\\RabbitMQ\\rabbitmq-stream-dotnet-client-main\\RabbitMQ.Stream.Client\\RoutingClient.cs:line 69\r\n   at RabbitMQ.Stream.Client.RoutingHelper`1.<LookupLeaderConnection>d__3.MoveNext() in C:\\Users\\kent.allen\\OneDrive - CentralSquare Technologies\\Desktop\\RabbitMQ\\rabbitmq-stream-dotnet-client-main\\RabbitMQ.Stream.Client\\RoutingClient.cs:line 133\r\n   at RabbitMQ.Stream.Client.RawProducer.<Create>d__8.MoveNext() in C:\\Users\\kent.allen\\OneDrive - CentralSquare Technologies\\Desktop\\RabbitMQ\\rabbitmq-stream-dotnet-client-main\\RabbitMQ.Stream.Client\\RawProducer.cs:line 62\r\n   at RabbitMQ.Stream.Client.StreamSystem.<CreateRawProducer>d__14.MoveNext() in C:\\Users\\kent.allen\\OneDrive - CentralSquare Technologies\\Desktop\\RabbitMQ\\rabbitmq-stream-dotnet-client-main\\RabbitMQ.Stream.Client\\StreamSystem.cs:line 245\r\n   at RabbitMQ.Stream.Client.Reliable.ProducerFactory.<StandardProducer>d__4.MoveNext() in C:\\Users\\kent.allen\\OneDrive - CentralSquare Technologies\\Desktop\\RabbitMQ\\rabbitmq-stream-dotnet-client-main\\RabbitMQ.Stream.Client\\Reliable\\ProducerFactory.cs:line 63\r\n   at RabbitMQ.Stream.Client.Reliable.ProducerFactory.<CreateProducer>d__2.MoveNext() in C:\\Users\\kent.allen\\OneDrive - CentralSquare Technologies\\Desktop\\RabbitMQ\\rabbitmq-stream-dotnet-client-main\\RabbitMQ.Stream.Client\\Reliable\\ProducerFactory.cs:line 29\r\n   at RabbitMQ.Stream.Client.Reliable.Producer.<CreateNewEntity>d__7.MoveNext() in C:\\Users\\kent.allen\\OneDrive - CentralSquare Technologies\\Desktop\\RabbitMQ\\rabbitmq-stream-dotnet-client-main\\RabbitMQ.Stream.Client\\Reliable\\Producer.cs:line 124\r\n   at RabbitMQ.Stream.Client.Reliable.ReliableBase.<Init>d__6.MoveNext() in C:\\Users\\kent.allen\\OneDrive - CentralSquare Technologies\\Desktop\\RabbitMQ\\rabbitmq-stream-dotnet-client-main\\RabbitMQ.Stream.Client\\Reliable\\ReliableBase.cs:line 65"

**Comment by Gsantomaggio** [comment]
I think that this is more of a windows problem not related to the client. See:
https://stackoverflow.com/questions/10461257/an-attempt-was-made-to-access-a-socket-in-a-way-forbidden-by-its-access-permissi

**Comment by kallencst** [comment, marked as answer]
I was able to resolve the problem. I had a VPN connection that was live. I had tried disconnecting, but this resulted in an exception. Eventually I created a new user and used this in place of "guest", and this allows the code to run when the VPN is disconnected. The issue occurs in RoutingClient.LookupConnection, when the VPN is connected, Dns.GetHostEntryAsync will return the VPN IP address as the first entry in the Address List, at least for me this is an IPV6 address that won't work when trying to connect a socket in Connection.Create. The code works with the "guest" user if you bypass the Dns.GetHostEntryAsync call.

**Reply by Gsantomaggio** [reply]
Ok please read also: https://github.com/rabbitmq/rabbitmq-stream-dotnet-client#load-balancer 
To connect the client with a Load-balancer/proxy.
